packages/synapse-react-client/src/components/HasAccess/HasAccessV2.tsx

```diff
@@ -126,6 +142,40 @@ export function useGetRestrictionUiType(
   return RestrictionUiType.AccessBlockedToAnonymous
 }
 
+/**
+ * This hook determines if
+ * - the entity is a FileEntity, AND
+ * - the caller has permission to fetch the dataFileHandle, AND
+ * - the dataFileHandle is an instance of ExternalFileHandleInterface (i.e. the file is not controlled by Synapse)
+ * Note that this requires an additional API call that cannot be batched, so it should be avoided in bulk contexts if possible.
+ */
+function useGetIsExternalFileHandle(
+  entityId: string,
+  options: { enabled: boolean },
+) {
+  const { data: entityBundle, isLoading } = useGetEntityBundle(
+    entityId,
+    undefined,
+    undefined,
+    {
+      ...options,
+      select: bundle => {
+        if (!isFileEntity(bundle.entity)) {
+          return false
+        }
+        const fileEntity = bundle.entity as FileEntity
+        const fileHandles = bundle.fileHandles as FileHandle[]
+        const fileHandle: FileHandle | undefined = fileHandles?.find(
+          fileHandle => fileHandle.id === fileEntity.dataFileHandleId,
+        )
+        return fileHandle && implementsExternalFileHandleInterface(fileHandle)
+      },
+    },
+  )
+
+  return { isExternalFileHandle: entityBundle, isLoading }
```

Comment:
Just return the entire `UseQueryResult`
```suggestion
  return useGetEntityBundle(
    entityId,
    undefined,
    undefined,
    {
      ...options,
      select: bundle => {
        if (!isFileEntity(bundle.entity)) {
          return false
        }
        const fileEntity = bundle.entity as FileEntity
        const fileHandles = bundle.fileHandles as FileHandle[]
        const fileHandle: FileHandle | undefined = fileHandles?.find(
          fileHandle => fileHandle.id === fileEntity.dataFileHandleId,
        )
        return fileHandle && implementsExternalFileHandleInterface(fileHandle)
      },
    },
  )
```